### PR TITLE
hpmv hpr hpr2 ILP64

### PR DESCRIPTION
**Pull request by NaveenElumalaiAMD**
resolves: [LWPMLSE-595](https://ontrack-internal.amd.com/browse/LWPMLSE-595)

### CODE REVIEW

File: clients/include/blas2/testing_hpmv.hpp [hunk 1]
```diff
@@ -78,50 +81,62 @@ void testing_hpmv_bad_arg(const Arguments& arg)
...
-                hipblasHpmvFn(handle, uplo, N, zero, nullptr, nullptr, incx, one, nullptr, incy));
+            DAPI_CHECK(hipblasHpmvFn,
+                       (handle, uplo, N, zero, nullptr, nullptr, incx, one, nullptr, incy));
+        }
+
+        // Check 64-bit API with quick return
+        if(arg.api & c_API_64)
+        {
+            DAPI_CHECK(hipblasHpmvFn, (handle, uplo, N, zero, dA, dx, incx, one, dy, incy));
         }
```
Comment (TorreZuk) [inline clients/include/blas2/testing_hpmv.hpp]: Why is this separate ?  Did you intend to set larger N ?

File: clients/include/blas2/testing_hpr2.hpp [hunk 2]
```diff
@@ -72,43 +74,54 @@ void testing_hpr2_bad_arg(const Arguments& arg)
...
-            CHECK_HIPBLAS_ERROR(
-                hipblasHpr2Fn(handle, uplo, N, zero, nullptr, incx, nullptr, incy, nullptr));
+            DAPI_CHECK(hipblasHpr2Fn,
+                       (handle, uplo, N, zero, nullptr, incx, nullptr, incy, nullptr));
+        }
+
+        // Check 64-bit API with quick return
+        if(arg.api & c_API_64)
+        {
+            DAPI_CHECK(hipblasHpr2Fn, (handle, uplo, N, zero, dx, incx, dy, incy, dA));
```
Comment (TorreZuk) [inline clients/include/blas2/testing_hpr2.hpp]: Also not sure here, same question on N

File: clients/include/blas2/testing_hpr2_strided_batched.hpp [hunk 3]
```diff
@@ -78,153 +83,186 @@ void testing_hpr2_strided_batched_bad_arg(const Arguments& arg)
...
+                         strideA,
+                         batch_count));
+        }
+
+        // Check 64-bit API with quick return
+        if(arg.api & c_API_64)
+        {
+            DAPI_CHECK(hipblasHpr2StridedBatchedFn,
+                       (handle,
+                        uplo,
```
Comment (TorreZuk) [inline clients/include/blas2/testing_hpr2_strided_batched.hpp:217]: ???

File: clients/include/blas2/testing_hpr_strided_batched.hpp [hunk 4]
```diff
@@ -76,60 +80,68 @@ void testing_hpr_strided_batched_bad_arg(const Arguments& arg)
...
+
+            DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                        hipblasHprStridedBatchedFn,
+                        (handle, uplo, N, alpha, dx, incx, stridex, nullptr, strideA, batch_count));
+        }
+        // Check 64-bit API with quick return
+        if(arg.api & c_API_64)
+        {
+            DAPI_CHECK(hipblasHprStridedBatchedFn,
+                       (handle, uplo, N, zero, dx, incx, stridex, dA, strideA, batch_count));
```
Comment (TorreZuk) [inline clients/include/blas2/testing_hpr_strided_batched.hpp]: ???

**Review by TorreZuk — APPROVED** [review]
Just those questions on the bad_args tests which I don't understand their purpose.

File: clients/include/blas2/testing_hpmv.hpp (hunk 1 above)
Comment (NaveenElumalaiAMD) [reply clients/include/blas2/testing_hpmv.hpp]: This was intended to just call the 64 bit API and return success.

Comment (TorreZuk) [reply clients/include/blas2/testing_hpmv.hpp]: > This was intended to just call the 64 bit API and return success.

All the DAPI_ calls call the 64bit API, so this case would be covered by the previous call above it already, no?

Comment (NaveenElumalaiAMD) [reply clients/include/blas2/testing_hpmv.hpp]: Oh yeah you are right, I somehow missed it. In that case I could do a larger N and expect a invalid value status just for AMD backend.

Comment (TorreZuk) [reply clients/include/blas2/testing_hpmv.hpp]: Well I would leave those to just be in rocBLAS only testing if you mean the thresholds for support.

**Review by daineAMD — APPROVED** [review]
Needs docs/ update, otherwise looks good. You can take a look at https://github.com/ROCm/hipBLAS/pull/805 to see what I had in mind for testing the 64-bit interface sizes to see if you like it.

**Pull request merged**